py/server/tests/test_pt_proxy.py

```diff
@@ -0,0 +1,300 @@
+#
+# Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+#
+
+import unittest
+
+from deephaven import read_csv, empty_table, SortDirection, AsOfMatchRule, DHError
+from deephaven.agg import sum_, avg, pct, weighted_avg, formula, group, first, last, max_, median, min_, std, abs_sum, \
+    var
+from deephaven.table import PartitionedTableProxy
+from tests.testbase import BaseTestCase
+
+
+class PartitionedTableProxyTestCase(BaseTestCase):
```

Comment:
If I'm correct about the `is_refreshing` attribute, you may need more testing to ensure that the lock is being acquired when it should be.